packages/shared-core/src/filters.ts

```diff
@@ -390,24 +390,42 @@ export const runLuceneQuery = (docs: any[], query?: SearchQuery) => {
     }
   )
 
-  // Match a document against all criteria
   const docMatch = (doc: any) => {
-    return (
-      stringMatch(doc) &&
-      fuzzyMatch(doc) &&
-      rangeMatch(doc) &&
-      equalMatch(doc) &&
-      notEqualMatch(doc) &&
-      emptyMatch(doc) &&
-      notEmptyMatch(doc) &&
-      oneOf(doc) &&
-      contains(doc) &&
-      containsAny(doc) &&
-      notContains(doc)
-    )
+    const filterFunctions: Record<SearchQueryOperators, (doc: any) => boolean> =
+      {
+        string: stringMatch,
+        fuzzy: fuzzyMatch,
+        range: rangeMatch,
+        equal: equalMatch,
+        notEqual: notEqualMatch,
+        empty: emptyMatch,
+        notEmpty: notEmptyMatch,
+        oneOf: oneOf,
+        contains: contains,
+        containsAny: containsAny,
+        notContains: notContains,
+      }
+
+    const activeFilterKeys: SearchQueryOperators[] = Object.entries(query || {})
+      .filter(
+        ([key, value]: [string, any]) =>
+          !["allOr", "onEmptyFilter"].includes(key) &&
+          Object.keys(value as Record<string, any>).length > 0
+      )
+      .map(([key]) => key as any)
+
+    const results: boolean[] = activeFilterKeys.map(filterKey => {
+      const filterFunction = filterFunctions[filterKey]
+      return filterFunction ? filterFunction(doc) : true
```

Comment:
I don't think we should return `true` for a missing filter function here - that would mean that if I sent up an invalid payload of `{ foo: { a : 1 } }` it will fail to find a filter function for `foo`, return `true` and therefore always match an `allOr` query because there's a true result, even if all my other valid filters fail.

I think we should return `false` so that it's completely ignored for a `allOr` query, and so that an "and" query will fail (which it should because this invalid filter function cannot match any rows).

You can also rewrite this as `return filterFunctions[filterKey]?.(doc) ?? false` to be a bit terser too.